fiat-web/src/main/java/com/netflix/spinnaker/fiat/config/ResourcesConfig.java

```diff
@@ -35,45 +30,18 @@
 import org.springframework.context.annotation.Primary;
 import org.springframework.context.annotation.PropertySource;
 import org.springframework.context.annotation.Scope;
-import retrofit.Endpoints;
-import retrofit.RestAdapter;
-import retrofit.converter.JacksonConverter;
 
 @Configuration
 @EnableConfigurationProperties(ProviderCacheConfig.class)
 @PropertySource("classpath:resilience4j-defaults.properties")
+@RequiredArgsConstructor
 public class ResourcesConfig {
-  @Autowired @Setter private RestAdapter.LogLevel retrofitLogLevel;
-
-  @Autowired @Setter private ObjectMapper objectMapper;
-
-  @Autowired @Setter private OkHttpClientProvider clientProvider;
-
-  @Value("${services.front50.base-url}")
-  @Setter
-  private String front50Endpoint;
-
-  @Value("${services.clouddriver.base-url}")
-  @Setter
-  private String clouddriverEndpoint;
-
-  @Value("${services.igor.base-url}")
-  @Setter
-  private String igorEndpoint;
+  private final ServiceClientProvider provider;
 
   @Bean
-  Front50Api front50Api() {
-    return new RestAdapter.Builder()
-        .setEndpoint(Endpoints.newFixedEndpoint(front50Endpoint))
-        .setClient(
-            new Ok3Client(
-                clientProvider.getClient(new DefaultServiceEndpoint("front50", front50Endpoint))))
-        .setConverter(new JacksonConverter(objectMapper))
-        .setErrorHandler(SpinnakerRetrofitErrorHandler.getInstance())
-        .setLogLevel(retrofitLogLevel)
-        .setLog(new Slf4jRetrofitLogger(Front50Api.class))
-        .build()
-        .create(Front50Api.class);
+  Front50Api front50Api(@Value("${services.front50.base-url}") String front50Endpoint) {
+    return provider.getService(
```

Comment:
I believe this ends up with [RetrofitServiceFactory](https://github.com/spinnaker/kork/blob/825c81a177d5c3b11a286d664759ea2786a4bd87/kork-retrofit/src/main/java/com/netflix/spinnaker/kork/retrofit/RetrofitServiceFactory.java#L51) creating the returned object.  At the moment that code doesn't use
```
.setErrorHandler(SpinnakerRetrofitErrorHandler.getInstance())
```
so this ends up being a change in behavior.  Adding the setErrorHandler call to RetrofitServiceFactory is in the plans, but is gonna take some care (and potentially a config flag controlling whether it happens or not) since I believe both gate and echo use it, and we need to coordinate changes in handling of RetrofitError to Spinnaker*Exception.

Reply:
Isn't the new error handler for running in Retrofit2? Right now, `ServiceClientFactory` implementations check for the use of the `Call<T>` API for determining which Retrofit version to use.

Reply:
SpinnakerRetrofitErrorHandler is for retrofit1.

Reply:
But Retrofit only throws `RetrofitError` in the first place?

Reply:
Yup, retrofit1 throws RetrofitError, and SpinnakerRetrofitErrorHandler turns RetrofitError into SpinnakerServerException (and children).

Reply:
Oh, so the point of the error handler Bean is to translate the retrofit exception into a spinnaker one?

Reply:
It's not a bean, but yes.